pkg/aws/client/client.go

```diff
@@ -458,20 +451,71 @@ func (c *Client) CreateBucketIfNotExists(ctx context.Context, bucket, region str
 		},
 	}
 
-	_, err = c.S3.PutBucketLifecycleConfiguration(ctx, putBucketLifecycleConfigurationInput)
+	if _, err = c.S3.PutBucketLifecycleConfiguration(ctx, putBucketLifecycleConfigurationInput); err != nil {
+		return err
+	}
+
+	if backupbucketConfig != nil && backupbucketConfig.Immutability != nil {
+		// update the bucket with object lock settings.
+		// note: object versioning is automatically get enabled if bucket is created with object lock.
+		return c.UpdateBucket(ctx, bucket, backupbucketConfig, true)
```

Comment:
Right now the `createBucket` not only creates the bucket but also immediately calls UpdateBucket. That mixes two responsibilities in one function. Could we split those into separate steps

Reply:
> That mixes two responsibilities in one function

nopes, `createBucket` with object lock enabled is actually itself a 2 step process:
you can check this : https://github.com/gardener/etcd-backup-restore/blob/master/docs/usage/enabling_immutable_snapshots.md#create-s3-buckets-with-object-lock-enabled

> Could we split those into separate steps

for reconciliation logic it should be single step process.

Reply:
I understand that enabling versioning and object lock involves multiple API calls. However, the name `updateBucket` might be too generic and could benefit from a more descriptive alternative that clearly reflects the specific configurations being applied. For instance, names like `configureBucketVersioningAndLock` or `enableBucketVersioningAndObjectLock` would more accurately convey the function's purpose.

Reply:
👍